### PR TITLE
provider: Support default tags (resources aws_c*)

### PR DESCRIPTION
**Pull request by bflad**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Reference: https://github.com/hashicorp/terraform-provider-aws/issues/7926

Output from acceptance testing: Will be handled via CI, review, and daily acceptance testing

### CODE REVIEW

File: aws/resource_aws_cognito_user_pool.go [hunk 1]
```diff
@@ -414,7 +414,8 @@ func resourceAwsCognitoUserPool() *schema.Resource {
 					},
 				},
 			},
-			"tags": tagsSchema(),
+			"tags":     tagsSchema(),
+			"tags_all": tagsSchemaComputed(),
```
Comment (anGie44) [inline aws/resource_aws_cognito_user_pool.go:418]: just needs the changes in `UserPoolUpdate`

**Review by anGie44 — APPROVED** [review]
Just a change needed in the user_pool resource + rebase, otherwise LGTM

**Comment by bflad** [comment]
Good catch on `aws_cognito_user_pool` -- updated and verified since the logic is a little different than other resources:

Output from acceptance testing:

```
--- PASS: TestAccAWSCognitoUserPool_basic (32.72s)
--- PASS: TestAccAWSCognitoUserPool_disappears (23.07s)
--- PASS: TestAccAWSCognitoUserPool_MfaConfiguration_SmsConfiguration (90.95s)
--- PASS: TestAccAWSCognitoUserPool_MfaConfiguration_SmsConfigurationAndSoftwareTokenMfaConfiguration (81.96s)
--- PASS: TestAccAWSCognitoUserPool_MfaConfiguration_SmsConfigurationToSoftwareTokenMfaConfiguration (52.93s)
--- PASS: TestAccAWSCognitoUserPool_MfaConfiguration_SoftwareTokenMfaConfiguration (46.95s)
--- PASS: TestAccAWSCognitoUserPool_MfaConfiguration_SoftwareTokenMfaConfigurationToSmsConfiguration (47.99s)
--- PASS: TestAccAWSCognitoUserPool_recovery (49.48s)
--- PASS: TestAccAWSCognitoUserPool_schemaAttributes (52.07s)
--- PASS: TestAccAWSCognitoUserPool_schemaAttributesModified (32.71s)
--- PASS: TestAccAWSCognitoUserPool_schemaAttributesRemoved (32.71s)
--- PASS: TestAccAWSCognitoUserPool_SmsAuthenticationMessage (45.81s)
--- PASS: TestAccAWSCognitoUserPool_SmsConfiguration (80.39s)
--- PASS: TestAccAWSCognitoUserPool_SmsConfiguration_ExternalId (75.83s)
--- PASS: TestAccAWSCognitoUserPool_SmsConfiguration_SnsCallerArn (73.17s)
--- PASS: TestAccAWSCognitoUserPool_SmsVerificationMessage (53.22s)
--- PASS: TestAccAWSCognitoUserPool_update (86.20s)
--- PASS: TestAccAWSCognitoUserPool_withAdminCreateUserConfiguration (43.93s)
--- PASS: TestAccAWSCognitoUserPool_withAdminCreateUserConfigurationAndPasswordPolicy (24.82s)
--- PASS: TestAccAWSCognitoUserPool_withAdvancedSecurityMode (69.48s)
--- PASS: TestAccAWSCognitoUserPool_withAliasAttributes (52.96s)
--- PASS: TestAccAWSCognitoUserPool_withDeviceConfiguration (43.98s)
--- PASS: TestAccAWSCognitoUserPool_withEmailConfiguration (28.09s)
--- PASS: TestAccAWSCognitoUserPool_withEmailVerificationMessage (44.74s)
--- PASS: TestAccAWSCognitoUserPool_withLambdaConfig (79.27s)
--- PASS: TestAccAWSCognitoUserPool_withPasswordPolicy (53.72s)
--- PASS: TestAccAWSCognitoUserPool_withTags (70.01s)
--- PASS: TestAccAWSCognitoUserPool_withUsernameConfiguration (53.49s)
--- PASS: TestAccAWSCognitoUserPool_withVerificationMessageTemplate (53.61s)
--- SKIP: TestAccAWSCognitoUserPool_withEmailConfigurationSource (0.00s)
```

File: aws/resource_aws_cloudwatch_metric_alarm.go [hunk 2]
```diff
@@ -237,8 +237,11 @@ func resourceAwsCloudWatchMetricAlarm() *schema.Resource {
 				ValidateFunc: validation.StringInSlice([]string{"evaluate", "ignore"}, true),
 			},
 
-			"tags": tagsSchema(),
+			"tags":     tagsSchema(),
+			"tags_all": tagsSchemaComputed(),
```
Comment (anGie44) [inline aws/resource_aws_cloudwatch_metric_alarm.go:241]: missed during first review @bflad : the `Create` method needs to pass in the new merged tags via nested func  `getAwsCloudWatchPutMetricAlarmInput`

File: aws/resource_aws_config_aggregate_authorization.go [hunk 3]
```diff
@@ -38,8 +38,11 @@ func resourceAwsConfigAggregateAuthorization() *schema.Resource {
 				Required: true,
 				ForceNew: true,
 			},
-			"tags": tagsSchema(),
+			"tags":     tagsSchema(),
```
Comment (anGie44) [inline aws/resource_aws_config_aggregate_authorization.go:41]: missed on first review: `resourceAwsConfigAggregateAuthorizationPut` needs to be updated as well to pass in the new merged tags

File: aws/resource_aws_cloudwatch_metric_alarm.go (hunk 2 above)
Comment (bflad) [reply aws/resource_aws_cloudwatch_metric_alarm.go:241]: Updated in https://github.com/hashicorp/terraform-provider-aws/pull/18728/commits/11eb3f1b23837bfa201117bdaa19d8354795e2e3

File: aws/resource_aws_config_aggregate_authorization.go (hunk 3 above)
Comment (bflad) [reply aws/resource_aws_config_aggregate_authorization.go:41]: Updated in https://github.com/hashicorp/terraform-provider-aws/pull/18728/commits/11eb3f1b23837bfa201117bdaa19d8354795e2e3

**Pull request merged**